lib/core-integration/src/Test/Integration/Scenario/API/ByronWallets.hs

```diff
@@ -0,0 +1,478 @@
+import Test.Hspec
+    ( SpecWith, describe, it )
+import Test.Integration.Framework.DSL
+    ( Context (..)
+    , Headers (..)
+    , Payload (..)
+    , emptyWallet
+    , expectErrorMessage
+    , expectResponseCode
+    , json
+    , listByronWalletEp
+    , postByronWalletEp
+    , request
+    , verify
+    , walletId
+    )
+import Test.Integration.Framework.TestData
+    ( arabicWalletName
+    , errMsg405
+    , errMsg406
+    , errMsg415
+    , frenchMnemonics12
+    , invalidMnemonics12
+    , japaneseMnemonics12
+    , kanjiWalletName
+    , mnemonics12
+    , mnemonics15
+    , mnemonics18
+    , mnemonics21
+    , mnemonics24
+    , mnemonics3
+    , mnemonics6
+    , mnemonics9
+    , passphraseMaxLength
+    , passphraseMinLength
+    , polishWalletName
+    , russianWalletName
+    , specMnemonicByron
+    , wildcardsWalletName
+    )
+
+import qualified Data.Text as T
+import qualified Network.HTTP.Types.Status as HTTP
+
+spec :: forall t. (EncodeAddress t, DecodeAddress t) => SpecWith (Context t)
+spec = do
+
+    it "BYRON_GET_02 - Byron ep does not show new wallet" $ \ctx -> do
+        w <- emptyWallet ctx
+        let ep  = ( "GET", "v2/byron/wallets/" <> w ^. walletId )
+        r <- request @ApiByronWallet ctx ep Default Empty
+        expectResponseCode @IO HTTP.status501 r
+
+    it "BYRON_LIST_02 - Byron ep does not list new wallets" $ \ctx -> do
+        _ <- emptyWallet ctx
+        r <- request @ApiByronWallet ctx listByronWalletEp Default Empty
+        expectResponseCode @IO HTTP.status501 r
+
+    it "BYRON_DELETE_02 - Byron ep does not delete new wallet" $ \ctx -> do
+        w <- emptyWallet ctx
+        let ep  = ( "DELETE", "v2/byron/wallets/" <> w ^. walletId )
+        r <- request @ApiByronWallet ctx ep Default Empty
+        expectResponseCode @IO HTTP.status501 r
+
+    it "BYRON_RESTORE_01 - Restore a wallet" $ \ctx -> do
+        mnemonic <- mnemonicToText @12 . entropyToMnemonic <$> genEntropy
+        let payload = Json [json| {
+                "name": "Empty Byron Wallet",
+                "mnemonic_sentence": #{mnemonic},
+                "passphrase": "Secure Passphrase"
+            }|]
+        r <- request @ApiByronWallet ctx postByronWalletEp Default payload
+        expectResponseCode @IO HTTP.status501 r
+
+        rg <- request @ApiByronWallet ctx listByronWalletEp Default Empty
+        expectResponseCode @IO HTTP.status501 rg
+
+    it "BYRON_RESTORE_02 - One can restore previously deleted wallet" $ \_ -> do
+        m <- mnemonicToText @12 . entropyToMnemonic <$> genEntropy
+        print m
+        -- w <- emptyByronWalletWith ctx ("Byron Wallet", m, "Secure Passphrase")
+        --
+        -- rd <- request @ApiByronWallet ctx (deleteByronWalletEp w ^. walletId) Default Empty
+        -- expectResponseCode @IO HTTP.status204 rd
+        --
+        -- wr <- emptyByronWalletWith ctx ("Byron Wallet2", m, "Secure Pa33phrase")
+        -- w ^. walletId `shouldBe` wr ^. walletId
```

Comment:
I'd rather not have commented out code committed to be honest, but I assume it's provisional.

Reply:
Yes. It's provisional.